api/users/views.py

```diff
@@ -229,6 +231,95 @@ def get_serializer_context(self):
         return context
 
 
+class UserAddonList(JSONAPIBaseView, generics.ListAPIView, ListFilterMixin, UserMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        CurrentUser,
+    )
+
+    serializer_class = UserAddonSettingsSerializer
+    view_category = 'users'
+    view_name = 'user-addons'
+
+    def get_queryset(self):
+        return self.get_user().get_addons()
+
+
+class UserAddonDetail(JSONAPIBaseView, generics.RetrieveAPIView, UserMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        CurrentUser,
+    )
+
+    serializer_class = UserAddonSettingsSerializer
+    view_category = 'users'
+    view_name = 'user-addon-detail'
+
+    def get_object(self):
+        user = self.get_user()
+        provider = self.kwargs['provider']
+        if provider not in ADDONS_OAUTH:
+            raise NotFound('Requested addon unavailable')
+
+        addon_settings = user.get_addon(provider)
+        if not addon_settings:
+            raise NotFound('Requested addon not enabled')
+        return addon_settings
+
+
+class UserAddonAccountList(JSONAPIBaseView, generics.ListAPIView, UserMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        CurrentUser,
+    )
+
+    serializer_class = AddonAccountSerializer
+    view_category = 'users'
+    view_name = 'user-external_accounts'
+
+    def get_queryset(self):
```

Comment:
Fixed in https://github.com/CenterForOpenScience/osf.io/pull/5532/commits/69477f1239f7ac7cc7e000436847d89c5b878661